libgit/browser.go

```diff
@@ -260,14 +264,14 @@ func (b *Browser) ReadDir(p string) (fis []os.FileInfo, err error) {
 		}
 	}
 
-	childrenPathsToCache := []string(nil)
+	childrenPathsToCache := make([]string, 0, len(dirTree.Entries))
 	for _, e := range dirTree.Entries {
-		fi, err := b.Lstat(e.Name)
+		fi, err := b.Lstat(path.Join(p, e.Name))
 		if err != nil {
 			return nil, err
 		}
 		fis = append(fis, fi)
-		childrenPathsToCache = append(childrenPathsToCache, path.Join(b.root, e.Name))
+		childrenPathsToCache = append(childrenPathsToCache, path.Join(b.root, p, e.Name))
```

Comment:
Can replace `b.root, p` with `cachePath`, since it already has them joined together.